pkg/docker/userns_remap.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package docker
+
+import (
+	"strings"
+
+	"sigs.k8s.io/kind/pkg/exec"
+)
+
+// UsernsRemap checks if userns-remap is enabled in dockerd
+func UsernsRemap() bool {
+	cmd := exec.Command("cat", "/etc/docker/daemon.json")
```

Comment:
I think we probably want to detect this from the daemon with something like `docker info`, I've not yet found the way to do this, but I suspect we can do something less brittle.

Reply:
I guess the only way to identify this via `docker info` is to check if the `Docker Root Dir` has in the end of the path something in the format of `UID.GID`. For example `/var/lib/docker/1000.1000`. 

Do you find this more safe?

Reply:
Aha! It seems that I found something in `docker info --format '{{json .SecurityOptions}}'`. Refactoring.